unittests/SubscriptionManagerTest.cpp

```diff
@@ -0,0 +1,880 @@
+    line.setFieldU32(ripple::sfFlags, ripple::lsfHighFreeze);
+    line.setFieldAmount(
+        ripple::sfBalance, ripple::STAmount(GetIssue(CURRENCY, ISSUER), 100));
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    EXPECT_CALL(*rawBackendPtr, doFetchLedgerObject).Times(3);
+    ON_CALL(*rawBackendPtr, doFetchLedgerObject)
+        .WillByDefault(Return(line.getSerializer().peekData()));
+    subManagerPtr->pubTransaction(trans1, ledgerinfo);
+    CheckSubscriberMessage(TransactionForOwnerFundFrozen, session);
+}
+
+/*
+ * test transaction for offer creation
+ * check owner_funds when issue global frozen
+ * mock backend return a frozen account setting
+ */
+TEST_F(
+    SubscriptionManagerSimpleBackendTest,
+    SubscriptionManagerTransactionOfferCreationGlobalFrozen)
+{
+    subManagerPtr->subTransactions(session);
+
+    auto ledgerinfo = CreateLedgerInfo(LEDGERHASH2, 33);
+    auto trans1 = TransactionAndMetadata();
+    ripple::STObject obj = CreateCreateOfferTransactionObject(
+        ACCOUNT1, 1, 32, CURRENCY, ISSUER, 1, 3);
+    trans1.transaction = obj.getSerializer().peekData();
+    trans1.ledgerSequence = 32;
+    ripple::STArray metaArray{0};
+    ripple::STObject metaObj(ripple::sfTransactionMetaData);
+    metaObj.setFieldArray(ripple::sfAffectedNodes, metaArray);
+    metaObj.setFieldU8(ripple::sfTransactionResult, ripple::tesSUCCESS);
+    metaObj.setFieldU32(ripple::sfTransactionIndex, 22);
+    trans1.metadata = metaObj.getSerializer().peekData();
+
+    ripple::STObject line(ripple::sfIndexes);
+    line.setFieldU16(ripple::sfLedgerEntryType, ripple::ltRIPPLE_STATE);
+    line.setFieldAmount(ripple::sfLowLimit, ripple::STAmount(10, false));
+    line.setFieldAmount(ripple::sfHighLimit, ripple::STAmount(100, false));
+    line.setFieldH256(ripple::sfPreviousTxnID, ripple::uint256{TXNID});
+    line.setFieldU32(ripple::sfPreviousTxnLgrSeq, 3);
+    line.setFieldU32(ripple::sfFlags, ripple::lsfHighFreeze);
+    auto issueAccount = GetAccountIDWithString(ISSUER);
+    line.setFieldAmount(
+        ripple::sfBalance, ripple::STAmount(GetIssue(CURRENCY, ISSUER), 100));
+    MockBackend* rawBackendPtr =
+        static_cast<MockBackend*>(mockBackendPtr.get());
+    EXPECT_CALL(*rawBackendPtr, doFetchLedgerObject).Times(2);
+    auto kk = ripple::keylet::account(issueAccount).key;
+    ON_CALL(
+        *rawBackendPtr, doFetchLedgerObject(testing::_, testing::_, testing::_))
+        .WillByDefault(Return(line.getSerializer().peekData()));
+    ripple::STObject accountRoot = CreateAccountRootObject(
+        ISSUER, ripple::lsfGlobalFreeze, 1, 10, 2, TXNID, 3);
+    ON_CALL(*rawBackendPtr, doFetchLedgerObject(kk, testing::_, testing::_))
+        .WillByDefault(Return(accountRoot.getSerializer().peekData()));
+    subManagerPtr->pubTransaction(trans1, ledgerinfo);
+    CheckSubscriberMessage(TransactionForOwnerFundFrozen, session);
+}
+
+/*
+ * test subscribe account
+ * */
```

Comment:
minor: looks like a different format of comments to what we use normally (compared to your other comments in this file)